srvutil/src/main/java/org/apache/rocketmq/srvutil/AclFileWatchService.java

```diff
@@ -0,0 +1,123 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.srvutil;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.rocketmq.common.ServiceThread;
+import org.apache.rocketmq.common.UtilAll;
+import org.apache.rocketmq.common.constant.LoggerName;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AclFileWatchService extends ServiceThread {
+    private static final InternalLogger log = InternalLoggerFactory.getLogger(LoggerName.COMMON_LOGGER_NAME);
+
+    private final String aclPath;
+    private int aclFilesNum;
+    private final Map<String, String> fileCurrentHash;
+    private final AclFileWatchService.Listener listener;
+    private static final int WATCH_INTERVAL = 500;
+    private MessageDigest md = MessageDigest.getInstance("MD5");
+
+    public AclFileWatchService(String path, final AclFileWatchService.Listener listener) throws Exception {
+        this.aclPath = path;
+        this.fileCurrentHash = new HashMap<>();
+        this.listener = listener;
+
+        File aclDir = new File(path);
+        String[] aclFileNames = aclDir.list();
+        this.aclFilesNum = aclFileNames.length;
+        for (int i = 0; i < aclFilesNum; i++) {
+            String aclFilePath = this.aclPath + aclFileNames[i];
+            if (StringUtils.isNotEmpty(aclFileNames[i]) && new File(aclFilePath).exists()) {
+                this.fileCurrentHash.put(aclFilePath, hash(aclFilePath));
+            }
+        }
+
+    }
+
+    @Override
+    public String getServiceName() {
+        return "AclFileWatchService";
+    }
+
+    @Override
+    public void run() {
+        log.info(this.getServiceName() + " service started");
+
+        while (!this.isStopped()) {
+            try {
+                this.waitForRunning(WATCH_INTERVAL);
+
+                File aclDir = new File(aclPath);
+                File[] aclFiles = aclDir.listFiles();
+                int realAclFilesNum = aclFiles.length;
+
+                if (aclFilesNum != realAclFilesNum) {
+                    log.info("aclFilesNum: " + aclFilesNum + "  realAclFilesNum: " + realAclFilesNum);
+                    aclFilesNum = realAclFilesNum;
+                    log.info("aclFilesNum: " + aclFilesNum + "  realAclFilesNum: " + realAclFilesNum);
+                    listener.onFileNumChanged(aclPath);
+                } else {
+                    for (int i = 0; i < aclFilesNum; i++) {
+                        String fileName = aclFiles[i].getAbsolutePath();
+                        String newHash = hash(fileName);
+                        if (!newHash.equals(fileCurrentHash.get(i))) {
```

Comment:
Should it be `fileCurrentHash.get(fileName)`?

Reply:
done